### PR TITLE
[AArch64] Add neon bici test for haddu and shadd

### PR DESCRIPTION
**Pull request by snikitav**
Add neon bici test for haddu and shadd, prerequisite for #76644

### CODE REVIEW

**Comment by snikitav** [comment]
@RKSimon @davemgreen PTAL

**Review by RKSimon — COMMENTED** [review]
Rename - these files are not about BICI they are about the HADD instructions

aarch64-neon-hadd.ll maybe?

**Comment by davemgreen** [comment]
Hello. I might be tempted to add known-bits to the test name to differentiate it from the other hadd tests we have, but either way this LGTM. Thanks

**Review by davemgreen — APPROVED** [review]

File: llvm/test/CodeGen/AArch64/aarch64-neon-hadd.ll [hunk 1]
```diff
@@ -0,0 +1,129 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 4
+; RUN: llc -mtriple=aarch64-neon < %s | FileCheck %s
```
Comment (davemgreen) [inline llvm/test/CodeGen/AArch64/aarch64-neon-hadd.ll:2]: -mtriple=aarch64 should be fine.

**Review by RKSimon — APPROVED** [review]
LGTM cheers

**Comment by davemgreen** [comment]
Thanks! LGTM

**Pull request merged**